game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/BattleStep.java

```diff
@@ -0,0 +1,88 @@
+package games.strategy.triplea.delegate.battle.steps;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.IExecutable;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.List;
+import lombok.AllArgsConstructor;
+
+/**
+ * A step in a battle.
+ *
+ * <ol>
+ *   <ul>
+ *     Each step can have 0 or more names. These names are shown in the {@link
+ *     games.strategy.triplea.ui.BattlePanel}
+ *   </ul>
+ *   <ul>
+ *     Each step can also have an executable. See {@link BattleAtomic} on what an executable is
+ *     comprised of.
+ *   </ul>
+ * </ol>
+ */
+@AllArgsConstructor
+public abstract class BattleStep {
+
+  /**
+   * The current state of the battle
+   *
+   * <p>Note: This is current mutable so the underlying data can change.
+   */
+  protected final BattleState battleState;
+
+  /** Actions that can occur in a battle that require interaction with {@link IDelegateBridge} */
+  protected final BattleActions battleActions;
+
+  public abstract BattleAtomic getExecutable();
+
+  public abstract List<String> getNames();
```

Comment:
A javadoc on this method could be of some use. It's a bit odd to think that a step would have multiple names. It'd be nice to know what it means to have multiple names for a step and secondarily if this is for display purposes only.

Reply:
The explanation about name is on the class.  I'll move it down to here.